src/node_config_file.cc

```diff
@@ -59,8 +59,12 @@ ParseResult ConfigReader::ParseNodeOptions(
             FPrintF(stderr, "Invalid value for %s\n", it->first.c_str());
             return ParseResult::InvalidContent;
           }
-          node_options_.push_back(it->first + "=" +
-                                  (result ? "true" : "false"));
+
+          if (result) {
```

Comment:
So if the user sets a flag to false the flag is not added. This means that if a flag is true by default and explicitly requires the value, that is ignored. We need to make sure this never happens. (Thats why I explicitly added `true` or `false`)

Reply:
First Id check if there is any flag with a boolean default value that requires an explicit boolean

Reply:
Okay, will do.

Reply:
```
node --experimental-strip-types=false

Welcome to Node.js v22.11.0.
Type ".help" for more information.
> (node:11533) ExperimentalWarning: Type Stripping is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)

> 
```